runtime/commits.go

```diff
@@ -44,13 +46,26 @@ func walkCommits(n *parser.NodeProgram, visitor *RuntimeVisitor) (*TableData, er
 		if boolRegister {
 			if !s.Count {
 				newRow := make(tableRow)
+
+				isNew := true
 				for _, f := range fields {
-					newRow[f] = metadataCommit(f, commit)
+					data := metadataCommit(f, commit)
+
+					if _, ok := seen[f]; !ok {
+						seen[f] = make(map[string]bool)
+					}
+
+					isNew = !seen[f][data]
+
+					newRow[f] = data
+					seen[f][data] = true
 				}
 
-				rows = append(rows, newRow)
+				if isNew || !s.Distinct {
+					counter = counter + 1
+					rows = append(rows, newRow)
+				}
 			}
-			counter = counter + 1
```

Comment:
Good and bad news 😅 
Good news: I found a problem trying to run `/gitql "select count(*) from commits " `. It returns 0 mainly because you removed this line. Can you revert that? 

Bad news: This is lack of functional tests,  hopefully I'm going to work on #92 soon. Despite that is all good for me.